### PR TITLE
ROE-449 Ensure register name has public prefixed

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-449

Updated "register_name" to "public_register_name" in corporate bo and would have done for corporate maaging officer but this is not present yet so updated it in the spec only.

### CODE REVIEW

**Comment by mwestacott** [comment]
Looks fine but presume we need to coordinate merging this with the corresponding change in the web/sdk project. Is that one ready to go to?

File: spec/schema.json [hunk 1]
```diff
@@ -486,7 +486,7 @@
         "is_on_register_in_country_formed_in": {
           "type": "boolean"
         },
-        "register_name": {
+        "public_register_name": {
```
Comment (mouhajer-ch) [inline spec/schema.json:489]: BeneficialOwnerGovernmentOrPublicAuthority does not have `public_register_name`, I think can be removed with `is_on_register_in_country_formed_in` and `registration_number`

Comment (mattch1) [reply spec/schema.json:489]: This is for ManagingOfficerCorporate

Comment (mouhajer-ch) [reply spec/schema.json:489]: Ops, you are right Mark! sorry mates

Comment (markpit) [reply spec/schema.json:489]: We'll have to look at BeneficialOwnerGovernmentOrPublicAuthority separately in another PR, I can see that there is no register_name or public_register_name in the spec for it, but there is also no field for it currently in the web model, but there appears to be a field for it in the v5 prototype.

Comment (mouhajer-ch) [reply spec/schema.json:489]: Nope, the register fields are on `beneficial-owner-other`, `managing-officer-corporate` and `entity`

Comment (markpit) [reply spec/schema.json:489]: Yes, you are right, I was looking at the corporate page! too confusing by half!

**Review by markpit — APPROVED** [review]

**Pull request merged**